src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -404,6 +418,42 @@ def _get_linux_fx_version(cmd, resource_group_name, name, slot=None):
     return site_config.linux_fx_version
 
 
+def url_validator(url):
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc, result.path])
+    except:
+        return False
+
+
+def _get_linux_multicontainer_config_file(cmd, resource_group_name, name, file_name=None, encoded=True, slot=None):
+    from base64 import b64encode, b64decode
+    config_file_bytes = None
+    if file_name:
+        if url_validator(file_name):
+            from urllib.request import urlopen
+            response = urlopen(file_name)
+            config_file_bytes = response.read()
+        else:
+            with open(file_name) as f:
+                file_contents = f.read()
+                config_file_bytes = file_contents.encode('utf-8')
+    else:
+        linux_fx_version = _get_linux_fx_version(cmd, resource_group_name, name, slot)
+        if not any([linux_fx_version.startswith(s) for s in MULTI_CONTAINER_TYPES]):
+            raise CLIError("Cannot decode config that is not one of the"
+                           " following types: {}".format(', '.join(MULTI_CONTAINER_TYPES)))
+        try:
+            config_file_bytes = b64decode(linux_fx_version.split('|')[1].encode('utf-8'))
+        except:
+            raise CLIError('Could not decode config')
```

Comment:
I suggest get rid of this if block all together, and just keep an string.
At high level,  I suggest  make the function do one thing which is to read file content, and not other things. This way the function will have a bit of readability. In details:
1. Please get rid of loops from encoding->decode->encoding again, and make this function just return a string
2. get rid of the `encoded` arguments, and let caller (only once place) to encode if needs